src/UiPath.CoreIpc/Helpers.cs

```diff
@@ -161,49 +160,78 @@ internal static async Task WriteMessage(this Stream stream, WireMessage message,
             if (data.Length == 0)
             {
                 return new();
             }
             return new(messageType, new MemoryStream(data));
         }
 
-        internal static async Task<byte[]> ReadBuffer(this Stream stream, int length, CancellationToken cancellationToken)
+        internal static async Task<byte[]> ReadBuffer(this Stream stream, int length, ILogger logger, CancellationToken cancellationToken)
         {
-            var messageData = await stream.ReadBufferCore(length, cancellationToken);
+            var messageData = await stream.ReadBufferCore(length, logger, cancellationToken);
             if (messageData.Length == 0)
             {
                 throw new IOException("Connection closed.");
             }
             return messageData;
         }
 
-        private static async Task<byte[]> ReadBufferCore(this Stream stream, int length, CancellationToken cancellationToken)
+        private static async Task<byte[]> ReadBufferCore(this Stream stream, int length, ILogger logger, CancellationToken cancellationToken)
         {
             var bytes = new byte[length];
             int offset = 0;
             int remaining = length;
-            while(remaining > 0)
+            
+            Stopwatch startRetry = new();
+
+            while (remaining > 0)
             {
-                var read = await stream.ReadAsync(bytes, offset, remaining, cancellationToken);
-                if(read == 0)
+                int read;
+                try
+                {
+                    read = await stream.ReadAsync(bytes, offset, remaining, cancellationToken);
+                }
+                catch (OperationCanceledException ex) when (stream is PipeStream && !cancellationToken.IsCancellationRequested)
+                {
+                    // In some Windows client environments, OperationCanceledException is sporadically thrown on named pipe ReadAsync operation (ERROR_OPERATION_ABORTED on overlapped ReadFile)
+                    // The cause has not yet been discovered(os specific, antiviruses, monitoring application), and we have implemented a retry system
+                    // ROBO-3083
+                    if (startRetry.IsRunning && startRetry.Elapsed < TimeSpan.FromSeconds(1))
+                    {
+                        // Throw for the second exception in the last 1 second
+                        logger?.LogException(ex, $"Second exception ReadAsync for {stream.GetType()}");
+                        throw;
+                    }
+                    else
+                    {
+                        startRetry.Restart();
+                        logger?.LogException(ex, $"Retrying ReadAsync for {stream.GetType()}");
+                        await Task.Delay(10); //Without this delay, on net framework can get OperationCanceledException on the second ReadAsync call
+                        continue;
+                    }
+                }
+                startRetry.Reset();
```

Comment:
**nit**: If it's on everyone's liking, it's safe to use `Stop` instead of `Reset`, because we always `Restart` and never `Start`.

`Stop` helps me personally to distinguish it from `Restart` and get a better understand of the algorithm.

Reply:
Done